pkg/apis/hive/v1/agent/platform.go

```diff
@@ -0,0 +1,131 @@
+package agent
+
+import (
+	//corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Platform stores cluster configuration for an agent based install where hosts are booted and registered, ready
+// to be assigned and used in a cluster.
+type Platform struct {
+	APIVIP            string `json:"apiVIP"`
+	APIVIPDNSName     string `json:"apiVIPDNSName"`
+	IngressVIP        string `json:"ingressVIP"`
+	VIPDHCPAllocation bool   `json:"vipDHCPAllocation"`
+
+	// Networking is the configuration for the pod network provider in
+	// the cluster.
+	Networking *Networking `json:"networking,omitempty"`
+
+	// Proxy defines the proxy settings for the cluster.
+	// If unset, the cluster will not be configured to use a proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	UserManagedNetworking  bool   `json:"userManagedNetworking,omitempty"`
+	AdditionalNtpSource    string `json:"additionalNtpSource,omitempty"`
```

Comment:
@avishayt what is this AdditionalNtpSource and how is it used? Is it something we're positive must be exposed on the kube API now? Is it day 2 configuration going to the cluster?

Reply:
It's a day1 configuration, we require to sync clocks between the hosts before we start the installation, so if user for some reason block access to a default ntp server we have a way for him to provide additional ntp sources.

Reply:
What's the format? IP address or hostname? Actually if you've got time Michael, if you could go through these in their current state and add godoc for each field, that would be super helpful for me as I don't know the source material very well.

Reply:
I'll try to do it tomorrow, it will a copy from what we have in the swagger file. 
https://github.com/openshift/assisted-service/blob/master/swagger.yaml#L3297-L3300

Reply:
In this case it can be both ip or name

Reply:
Ah no worries then, I can work off your swagger so I can add it here.

Reply:
why is this named additional ? Is this added to the existing RHCOS embeded one or does it overwrite to make this the only ntp source. If this is latter i think we should just rename to NTP.
also make it a struct since it seems like it will be more than one field like sources, frequency etc..

Reply:
That's correct. Added to any predefined NTP sources (could be received by DHCP, default configuration, or manually added)

Reply:
See https://github.com/openshift/enhancements/pull/574#discussion_r551417090

It's a very common requirement for users to need to be able to specify a different NTP source than the default - for example, AIUI we have no access to external NTP servers from within Red Hat and have to configure our installs to use an internal NTP server

So, while for agent-based installs we need to configure the pre-install machines the agent is running on, and also the machines post-install ... there is nothing specific to agent-based installation about needing to configure an NTP server

IMO this should be at the ClusterDeployment level, not platform-specific

Reply:
I agree, it's not platform-specific.

Reply:
Since we need this for the discovery ISO (so that clocks can be synchronized prior to starting installation), and since the discovery ISO will not be cluster-specific in the future, do we need this on the cluster definition at all? NTP will presumably be on the `InstallEnvironment` (or similar name) resource we've been discussing, and used there in the generation of an ISO.